### ISSUE TITLE
EAP CI seems to be failing regularly

### ISSUE THREAD

**Issue by patflynn** [GoogleCloudPlatform/google-cloud-intellij#1690]
https://kokoro.corp.google.com/job/google-cloud-intellij/job/gcp_ubuntu/job/eap/job/continuous/125/

**Comment by etanshaul** [comment]
@nkibler I'm seeing some kotlin.TypeCastException in the logs so it may be Kotlin related. Mind taking a look since you've looked into this before?

**Comment by nkibler** [comment]
The latest failure points to an issue within the SDK itself. It looks like that tests that extend from `PlatformTestCase` fail with the following stacktrace:

```
java.lang.ExceptionInInitializerError
        at com.intellij.openapi.vfs.newvfs.ManagingFS.getInstance(ManagingFS.java:36)
        at com.intellij.openapi.vfs.newvfs.persistent.PersistentFS.getInstance(PersistentFS.java:52)
        at com.intellij.testFramework.PlatformTestCase.l(PlatformTestCase.java:186)
        at com.intellij.testFramework.PlatformTestCase.initApplication(PlatformTestCase.java:148)
        at com.intellij.testFramework.PlatformTestCase.setUp(PlatformTestCase.java:211)
```

While the cause seems Kotlin related, it's simply because `picoContainer.getComponentInstance()` for the `ManagingFS` component is returning `null`, which fails the Kotlin-injected null-check during a cast to the `IComponentStore` interface.

Somehow, the `ManagingFS` component wasn't registered properly or has no instance bound in the `PicoContainer`. Still digging further into this so stay tuned.

**Comment by nkibler** [comment]
I haven't figured anything else out, but I believe this is a problem with the SDK. I filed this issue on their side: https://youtrack.jetbrains.com/issue/IJSDK-323

**Comment by etanshaul** [comment]
So it fails with a simple test that extends `PlatformTestCase` with no dependency on our plugin?

**Comment by nkibler** [comment]
Oh I'm not sure about that, I guess I would need to set up a new plugin project to determine that?

**Comment by etanshaul** [comment]
is this reproducible when tests are run individually? if so, I was thinking of just trying a simple test in our plugin to confirm that it still fails when the test has no code depending on our plugin. I assume it would from what you've written, but just checking.

**Comment by nkibler** [comment]
FYI, I also verified that it's also affecting tests that use the `CloudToolsRule`.

Here's an example command to run a single test:

```
./gradlew google-cloud-tools-plugin:test \
    --tests com.google.cloud.tools.intellij.resources.GoogleUserModelItemTest \
    -Dorg.gradle.project.intellijRepoUrl="https://www.jetbrains.com/intellij-repository" \
    -Dorg.gradle.project.ideaVersion="LATEST-EAP-SNAPSHOT" \
    --stacktrace
```

And here's the output for the command above:

```
com.google.cloud.tools.intellij.resources.GoogleUserModelItemTest > getImage_doesReturnUserImage FAILED
    java.lang.RuntimeException: java.lang.ClassNotFoundException: org.intellij.plugins.xpathView.XPathAppComponent

        Caused by:
        java.lang.ClassNotFoundException: org.intellij.plugins.xpathView.XPathAppComponent

com.google.cloud.tools.intellij.resources.GoogleUserModelItemTest > getCredentialedUser_doesReturnUser FAILED
    java.lang.ExceptionInInitializerError

        Caused by:
        kotlin.TypeCastException: null cannot be cast to non-null type com.intellij.openapi.components.impl.stores.IComponentStore

com.google.cloud.tools.intellij.resources.GoogleUserModelItemTest > getChildCount_withNotMatchingFilter_doesReturnFilteredChildren FAILED
    java.lang.NoClassDefFoundError: Could not initialize class com.intellij.openapi.vfs.LocalFileSystem$LocalFileSystemHolder

com.google.cloud.tools.intellij.resources.GoogleUserModelItemTest > getChildAt_withMatchingFilter_doesReturnFilteredChildren FAILED
    java.lang.NoClassDefFoundError: Could not initialize class com.intellij.openapi.vfs.LocalFileSystem$LocalFileSystemHolder

com.google.cloud.tools.intellij.resources.GoogleUserModelItemTest > getChildAt_withNotMatchingFilter_doesReturnFilteredChildren FAILED
    java.lang.NoClassDefFoundError: Could not initialize class com.intellij.openapi.vfs.LocalFileSystem$LocalFileSystemHolder

com.google.cloud.tools.intellij.resources.GoogleUserModelItemTest > getEmail_doesReturnUserEmail FAILED
    java.lang.NoClassDefFoundError: Could not initialize class com.intellij.openapi.vfs.LocalFileSystem$LocalFileSystemHolder

com.google.cloud.tools.intellij.resources.GoogleUserModelItemTest > newInstance_doesAddLoadingModelItem FAILED
    java.lang.NoClassDefFoundError: Could not initialize class com.intellij.openapi.vfs.LocalFileSystem$LocalFileSystemHolder

com.google.cloud.tools.intellij.resources.GoogleUserModelItemTest > getChildCount_withMatchingFilter_doesReturnFilteredChildren FAILED
    java.lang.NoClassDefFoundError: Could not initialize class com.intellij.openapi.vfs.LocalFileSystem$LocalFileSystemHolder

com.google.cloud.tools.intellij.resources.GoogleUserModelItemTest > getChildAt_withEmptyFilter_doesReturnAllChildren FAILED
    java.lang.NoClassDefFoundError: Could not initialize class com.intellij.openapi.vfs.LocalFileSystem$LocalFileSystemHolder

com.google.cloud.tools.intellij.resources.GoogleUserModelItemTest > getChildCount_withEmptyFilter_doesReturnAllChildren FAILED
    java.lang.NoClassDefFoundError: Could not initialize class com.intellij.openapi.vfs.LocalFileSystem$LocalFileSystemHolder

com.google.cloud.tools.intellij.resources.GoogleUserModelItemTest > getName_doesReturnUserName FAILED
    java.lang.NoClassDefFoundError: Could not initialize class com.intellij.openapi.vfs.LocalFileSystem$LocalFileSystemHolder

com.google.cloud.tools.intellij.resources.GoogleUserModelItemTest > isSynchronizing_doesReturnFalse FAILED
    java.lang.NoClassDefFoundError: Could not initialize class com.intellij.openapi.vfs.LocalFileSystem$LocalFileSystemHolder

12 tests completed, 12 failed
:google-cloud-tools-plugin:test FAILED
```

**Comment by nkibler** [comment]
And to your point, the following dummy test that only extends from `PlatformTestCase` and has nothing specific to our plugin:

```java
public final class EAPFailureTest extends PlatformTestCase {

  public void testSomething() {
    assertTrue(true);
  }
}
```

... fails with the following error:

```
com.google.cloud.tools.intellij.EAPFailureTest > testSomething FAILED
    java.lang.RuntimeException: java.lang.ClassNotFoundException: org.intellij.plugins.xpathView.XPathAppComponent

        Caused by:
        java.lang.ClassNotFoundException: org.intellij.plugins.xpathView.XPathAppComponent

1 test completed, 1 failed
:google-cloud-tools-plugin:test FAILED
```

So we know it's not our code. Could it be something with our Gradle task setup?

**Comment by etanshaul** [comment]
a shit. well you could quick clone another opensource plugin, add your EAPFailureTest to it and see what happens

**Comment by etanshaul** [comment]
EAP is now fixed via #1701 and JB fixing https://youtrack.jetbrains.com/issue/IJSDK-323
